### PR TITLE
Refactored network config (systemd-networkd)

### PR DESCRIPTION
**Pull request by mycognosist**
# Refactor network configuration to use `systemd-networkd`

We have experienced major stability issues with access point and client switching using `networking.service`, `hostapd` and `dnsmasq`. This refactor replaces the classic Debian networking configuration with `systemd-networkd` and allows switching of wireless mode without having to stop any services. This approach should bring much greater stability to PeachCloud networking.

We are following the steps outlined here: [Switch between wifi client and access point without reboot](https://raspberrypi.stackexchange.com/questions/93311/switch-between-wifi-client-and-access-point-without-reboot)

- Legacy configuration files have been moved to `conf/backup`.
- New network configuration files have been added to `conf/network`.
- Documentation for this new network configuration (as well as configuration steps for using a second wireless adapter) have been added to `docs`
- Network configuration script has been added to `scripts/setup_networking.py`

_Note:_ The main installation script (`scripts/setup_dev_env.py`) has not yet been updated, neither have the various documentation assets which describe the PeachCloud networking landscape.

-----

CC: @mhfowler - you're invited to test this new approach on your Pi:

```bash
cd peach-config
git pull
git checkout revised_network_config
sudo -Es
# edit `conf/network/wpa_supplicant-wlan0.conf` with your ap credentials (before running the script) or
# edit `/etc/wpa_supplicant/wpa_supplicant-wlan0.conf` with your ap credentials (after running the script)
python scripts/setup_networking.py
```

Client mode is activated on boot (by default). Switch to access point mode like this:

`sudo systemctl start wpa_supplicant@ap0.service`

And switch back to client mode like this:

`sudo systemctl start wpa_supplicant@wlan0.service`

It's that easy :)

You can then run various status checks:

```bash
# check interface statuses
ip a
sudo systemctl status wpa_supplicant@wlan0.service
sudo systemctl status wpa_supplicant@ap0.service
```

If you encounter any DNS / connectivity errors after the upgrade, have a look at the Troublshooting section in the README to ensure your system datetime is not causing an issue.

### CODE REVIEW

**Comment by mycognosist** [comment]
# Refactor `scripts/setup_dev_env.py` & update README

The following changes have been made to the primary setup script (`scripts/setup_dev_env.py`):

- Install `libnss-resolve`
- Remove installation of `hostapd`, `dnsmasq` and `avahi-daemon`
- Remove creation of `wpactrl-user`
- Remove creation of `peach-network` user
- Remove addition of `peach-network` user to `wpactrl-user` group
- Remove all networking configuration steps except copying `conf/hostname` and `conf/hosts`
- Remove copy of `network` sudoers rule file
- Add all configuration steps from `scripts/setup_networking.py`

Version number has been bumped to `0.2.4` and the `README` has been updated with the latest networking details.

-----

The `peach-network` microservice must now be updated inline with these changes:

- Update `peach-network.service` to run as `netdev:netdev`
- Update `add()` RPC to write to `/etc/wpa_supplicant/wpa_supplicant-wlan0.conf`
- Update `activate_ap()` and `activate_client()` RPCs
- Update documentation (also in `peach-devdocs`)

**Comment by mycognosist** [comment]
CC: @mhfowler - please read through the latest commit (especially the updated `README`), just to catch anything I might have missed. I'll merge once I hear back from you.

File: README.md [hunk 1]
```diff
@@ -64,6 +62,8 @@ nano /etc/network/interfaces
 auto lo
 iface lo inet loopback
 
+iface eth0 inet dhcp
```
Comment (mhfowler) [inline README.md:65]: we can consider to add 
```
auto eth0
allow-hotplug eth0
```
I believe I needed that for getting the ethernet to connect when I plugged it in to my mac which was sharing etherent, 
but thats really separate from this PR 
just noting it as you added this line

File: scripts/setup_dev_env.py [hunk 2]
```diff
@@ -130,27 +124,72 @@
...
-                 "/lib/systemd/system/networking.service.d/reduce-timeout.conf"])
-subprocess.call(["cp", "scripts/activate_ap.sh", "/usr/local/bin/activate_ap"])
-subprocess.call(["chmod", "755", "/usr/local/bin/activate_ap"])
-subprocess.call(["cp", "scripts/activate_client.sh",
-                 "/usr/local/bin/activate_client"])
-subprocess.call(["chmod", "755", "/usr/local/bin/activate_client"])
-# Allow group members to write to wpa_supplicant.conf
-subprocess.call(["chmod", "664", "/etc/wpa_supplicant/wpa_supplicant.conf"])
-# Set ownership so that wpa_supplicant.conf can be written to by peach-network
-subprocess.call(["chown", "root:wpactrl-user",
```
Comment (mhfowler) [inline scripts/setup_dev_env.py:152]: nice to get to remove so much

**Review by mhfowler — APPROVED** [review]
looks good, can't see any thing from the old config staying around 

my one comment was not really related to this PR and more of a question
about the eth0 interface

**Comment by mycognosist** [comment]
@mhfowler 

I've added your suggestion to the `README` for `eth0` (`auto` and `hotplug`). This doesn't seem to matter for `systemd-networkd` (the `/etc/network/interfaces` file is erased when `systemd-networkd` is setup) but at least it will enable hotplugging prior to the execution of the `setup_dev_env.py` and / or `setup_networking.py` scripts.

**Pull request merged**